### ISSUE TITLE
Supabase cli (edge functions) not working with OpenAI deno.land (but works with esm.sh)

### ISSUE THREAD

**Issue by ferrannp** [supabase/supabase#26875]
## Describe the bug

Code not working:

```js
import OpenAI from "https://deno.land/x/openai@v4.47.1/mod.ts";

const openai = new OpenAI({
  apiKey: "YOUR_API_KEY"
});
```

You get the following error:

```js
serving the request with /home/deno/functions/generate-workout
runtime has escaped from the event loop unexpectedly: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.1/lib/ChatCompletionStream.ts:4:14
failed to send request to user worker: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.1/lib/ChatCompletionStream.ts:4:14
user worker failed to respond: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.1/lib/ChatCompletionStream.ts:4:14
InvalidWorkerResponse: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.1/lib/ChatCompletionStream.ts:4:14
    at async Promise.allSettled (index 1)
    at async UserWorker.fetch (ext:sb_user_workers/user_workers.js:77:21)
    at async Object.handler (file:///home/deno/main/index.ts:165:14)
    at async respond (ext:sb_core_main_js/js/http.js:162:14) {
  name: "InvalidWorkerResponse"
```

HOWEVER, it works if you:

```js
import OpenAI from "https://esm.sh/openai@4.47.2";
```

## Expected behavior

It should work with deno.land.

## System information

- OS: macOS
- Version of supabase cli: 1.169.8
- Version of Node.js: 18.16.0

**Comment by encima** [comment]
Hey there, 

Thanks for opening!
Is this on hosted or running locally?
I cannot reproduce this when running locally and the example in the docs works when using the Deno package you outlined:

```
import OpenAI from 'https://deno.land/x/openai@v4.47.0/mod.ts'

Deno.serve(async (req) => {
  const { query } = await req.json()
  const apiKey = Deno.env.get('OPENAI_API_KEY')
  const openai = new OpenAI({
    apiKey: apiKey,
  })

  // Documentation here: https://github.com/openai/openai-node
  const chatCompletion = await openai.chat.completions.create({
    messages: [{ role: 'user', content: query }],
    // Choose model from here: https://platform.openai.com/docs/models
    model: 'gpt-3.5-turbo',
    stream: false,
  })

  const reply = chatCompletion.choices[0].message.content

  return new Response(reply, {
    headers: { 'Content-Type': 'text/plain' },
  })
})
```

Are you making sure that you create the client inside the `serve` function? If yes, can you provide a full reproducible code sample, please?

**Comment by ferrannp** [comment]
Hey @encima! So strange. Your code also errors for me locally:

```
Using supabase-edge-runtime-1.53.3 (compatible with Deno v1.40.3)
serving the request with /home/deno/functions/generate-workout
runtime has escaped from the event loop unexpectedly: event loop error: ReferenceError: _computedKey is not defined
    at https://deno.land/x/openai@v4.47.0/lib/ChatCompletionStream.ts:4:14
user worker failed to respond: user worker not available
TypeError: request body receiver not connected (channel closed)
    at Object.write (ext:deno_web/06_streams.js:1155:20)
    at Module.invokeCallbackFunction (ext:deno_webidl/00_webidl.js:944:16)
    at WritableStreamDefaultController.writeAlgorithm (ext:deno_web/06_streams.js:3915:14)
    at writableStreamDefaultControllerProcessWrite (ext:deno_web/06_streams.js:4485:55)
    at writableStreamDefaultControllerAdvanceQueueIfNeeded (ext:deno_web/06_streams.js:4388:5)
    at writableStreamDefaultControllerWrite (ext:deno_web/06_streams.js:4532:3)
    at writableStreamDefaultWriterWrite (ext:deno_web/06_streams.js:4679:3)
    at Object.chunkSteps (ext:deno_web/06_streams.js:2743:17)
    at readableStreamFulfillReadRequest (ext:deno_web/06_streams.js:2585:17)
    at readableByteStreamControllerCommitPullIntoDescriptor (ext:deno_web/06_streams.js:2276:5)
```

I have:

```
macOS: Sonoma 14.5
supabase-cli: 1.172.2
node: v18.16.0
```

**Comment by msgerbush** [comment]
Running into this same issue locally, if I downgrade the Supabase CLI to 1.166.3 it resolves the issue (it's broken starting at 1.167.0).

Looks related to an issue in the Deno runtime: https://github.com/denoland/deno/issues/22109

**Comment by laktek** [comment]
Fixed in https://github.com/supabase/edge-runtime/pull/352
